amethyst_ui/src/prefab.rs

```diff
@@ -282,3 +293,242 @@ where
         self.image.trigger_sub_loading(progress, &mut system_data.1)
     }
 }
+
+/// Loadable ui components
+///
+/// ### Type parameters:
+///
+/// - `I`: `Format` used for loading `Texture`s
+/// - `F`: `Format` used for loading `FontAsset`
+#[derive(Serialize, Deserialize)]
+pub enum UiWidget<I, F>
+where
+    I: Format<Texture, Options = TextureMetadata>,
+    F: Format<FontAsset, Options = ()>,
+{
+    /// Container component, have a transform, an optional background image, and children
+    Container(
+        UiTransformBuilder,
+        Option<UiImageBuilder<I>>,
+        Vec<UiWidget<I, F>>,
+    ),
+    /// Image component, have a transform and an image
+    Image(UiTransformBuilder, UiImageBuilder<I>),
+    /// Text component, have a transform and text
+    Text(UiTransformBuilder, UiTextBuilder<F>),
```

Comment:
Can we make these a struct-like variant? Looking at the RON file, the tuple style is a bit confusing at first. I generally think that tuples should be avoided in serialization except it's something that fits well (a coordinate for example, but nothing that's nested).

Reply:
I don't know how to make the separate variants then ?

Reply:
Unless you mean that the data inside the variant should be a struct?

Reply:
Just

```
Image {
    transform: ..,
    image: ..,
}
```

as far as I remember. So instead of a tuple variant, just a variant with fields.

Reply:
I will make it so